src/implementation/Server/DaprServer.ts

```diff
@@ -111,9 +99,26 @@ export default class DaprServer {
     }
   }
 
+  private getClientOptions(
+    clientoptions: Partial<DaprClientOptions> | undefined,
+    communicationProtocol: CommunicationProtocolEnum,
+  ): DaprClientOptions {
+    const clientCommunicationProtocol = clientoptions?.communicationProtocol ?? communicationProtocol;
+    return {
+      daprHost: clientoptions?.daprHost ?? Settings.getDefaultHost(),
+      daprPort: clientoptions?.daprPort ?? Settings.getDefaultAppPort(clientCommunicationProtocol),
+      communicationProtocol: clientCommunicationProtocol,
+      isKeepAlive: clientoptions?.isKeepAlive,
+      logger: clientoptions?.logger,
+      actor: clientoptions?.actor,
+      daprApiToken: clientoptions?.daprApiToken,
+      maxBodySizeMb: clientoptions?.maxBodySizeMb,
+    };
```

Comment:
Can we move this to a common place and use the same in DaprClient?

Reply:
Done